From 9bac97ee6616fc834a4da8d24df97714241b9d02 Mon Sep 17 00:00:00 2001
From: Hendrik Leppkes <h.leppkes@gmail.com>
Date: Fri, 20 Jul 2012 23:03:32 +0200
Subject: [PATCH 09/62] mpegvideo: null out current_picture_ptr when allocation
 failed

This avoids using an uninitialized picture in further decode calls.
---
 libavcodec/mpegvideo.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/libavcodec/mpegvideo.c b/libavcodec/mpegvideo.c
index 60e19ff..5f53087 100644
--- a/libavcodec/mpegvideo.c
+++ b/libavcodec/mpegvideo.c
@@ -1244,8 +1244,10 @@ int ff_mpv_frame_start(MpegEncContext *s, AVCodecContext *avctx)
 
     pic->f->coded_picture_number = s->coded_picture_number++;
 
-    if (alloc_picture(s, pic, 0) < 0)
+    if (alloc_picture(s, pic, 0) < 0) {
+        s->current_picture_ptr = NULL;
         return -1;
+    }
 
     s->current_picture_ptr = pic;
     // FIXME use only the vars from current_pic
-- 
2.6.3

